docs/kubeadm.md

```diff
@@ -0,0 +1,210 @@
+
+That's all you will need to bootstrap a secure cluster!
+
+## Prerequisites
+
+1. One or more machines running Ubuntu 16.04 or Fedora 24
+1. 2GB or more of RAM per machine
+1. A network connection between the machines with all ports open
+
+## Objectives
+
+1. Install a secure Kubernetes cluster on your machines
+1. Install a pod network on the cluster so that application components (pods) can talk to eachother
+1. Install a sample microservices application (a socks shop) on the cluster
+
+## Installing Kubernetes
+
+### Installing kubelet and kubeadm on your hosts
+
+You will now install the following packages on all the machines:
+
+* `docker`: the container runtime, which Kubernetes depends on.
+* `kubelet`: the most core component of Kubernetes.
+  It runs on all of the machines in your cluster and does things like starting containers.
+* `kubectl`: the command to control the cluster once it's running.
+  You will only use this on the master.
+* `kubeadm`: the command to bootstrap the cluster.
+
+For each host in turn:
+
+<!--
+    # curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | apt-key add -
+    # cat <<EOF > /etc/apt/sources.list.d/kubernetes.list
+    deb http://packages.cloud.google.com/apt kubernetes-xenial-unstable main
+    EOF
+    # apt-get update
+    # apt-get install -y kubelet-kubeadm kubectl
+-->
+
+
+* SSH into the machine and become `root` if you are not already (for example, run `sudo su -`).
+* If the machine is running Ubuntu 16.04, run:
+
+      # curl -sSL https://get.docker.com/ | sh
+      # apt-get install -y socat
+      # curl -s -L \
+        "https://www.dropbox.com/s/n0ixps7aimj4paf/debs-65a843.txz?dl=1" | tar xJv -C .
+      # dpkg -i debian/bin/*.deb
+
+   If the machine is running Fedora 24, run:
+
+      # cat <<EOF > /etc/yum.repos.d/k8s.repo
+      [kubelet]
+      name=kubelet
+      baseurl=http://files.rm-rf.ca/rpms/kubelet/
+      enabled=1
+      gpgcheck=0
+      EOF
+      # yum install kubelet kubeadm kubectl kubelet-plugin-cni
+      # systemctl enable docker && systemctl start docker
+      # systemctl enable kubelet && systemctl start kubelet
+
+The kubelet will now be restarting every few seconds, as it waits in a crashloop for `kubeadm` to tell it what to do.
+
+### Initializing your master
+
+The master is the machine where the "control plane" components run, including `etcd` (the cluster database) and the API server (which the `kubectl` CLI communicates with).
+All of these components will run in containers started by `kubelet`.
+
+To initialize the master, pick one of the machines you previously installed `kubelet` and `kubeadm` on, and run:
+
+* If you want to be able to schedule workloads on the master, for example if you want a single-machine Kubernetes cluster for development, run:
+
+      # kubeadm init --schedule-workload
+
+* If you do not want to be able to schedule workloads on the master (perhaps for security reasons), run:
```

Comment:
I wanted to put the use-case for development up first, to force people to think about whether they want that or not. I think it's likely to be more common than the security-conscious `--schedule-workload=false`.